### ISSUE TITLE
Bump `@sindresorhus/is` to latest

### ISSUE THREAD

**Issue by ikelax** [omnidan/node-emoji#205]
Currently, we have to use the `.default` before every property of `is`, because we're on an older version of `@sindresorhus/is` that doesn't set up ESM defaults properly yet. An example for this is

By bumping `@sindresorhus/is` to latest, we could avoid this.

_Originally posted by @JoshuaKGoldberg in https://github.com/omnidan/node-emoji/pull/204#discussion_r1791914321_

**Comment by JoshuaKGoldberg** [comment]
Agreed, that'd be great. The one tricky thing would be making sure it doesn't change Node.js support for `node-emoji` itself:

* Minimum Node.js version: not raising it
* CJS/ESM: not-old versions of `@sindresorthus/*` packages don't support CJS natively

...so if either of this being broken is a prerequisite for a version of `@sindresorhus/is` that fixes the `.default` shenanigan, then this'd be a breaking change for the next major version.
